packages/@lwc/engine-server/src/apis/render-component.ts

```diff
@@ -30,8 +30,9 @@ const FakeRootElement: HostElement = {
 export function renderComponent(
     tagName: string,
     Ctor: typeof LightningElement,
-    props: { [name: string]: any } = {}
-): string {
+    props: { [name: string]: any } = {},
+    options: SsrOptions = {}
```

Comment:
@priandsf What is your vision for using SSR rendered components and client components at the same time?

If a component in the SSR rendered tree has JS dependencies the JS will be ignored currently. 
I was imaging we may want to filter out which components are rendered on the server and which tags are preserved and delivered un-rendered.

Reply:
Long story short: yes. But this is out of the scope of this PR.